python/cudf/dataframe/column.py

```diff
@@ -17,10 +18,19 @@
 
 from cudf.utils import cudautils, utils, ioutils
 from cudf.dataframe.buffer import Buffer
+from cudf.dataframe import columnops
 from cudf.bindings.cudf_cpp import count_nonzero_mask, column_view_pointer
 from cudf.bindings.concat import _column_concat
 
 
+@cuda.jit
+def boolean_assign_kernel(data, mask, out):
+    idx = cuda.threadIdx.x + cuda.threadIdx.y * cuda.blockDim.x
+    # assume data.size == mask.size == out.size
+    if idx < data.size and mask[idx]:
```

Comment:
You can't use `mask[idx]` here as it will return you 8 bits (in the case of int8 dtype being used for the mask), whereas you're only looking for the specific bit at `idx`. Instead you should use `cudf.utils.utils.mask_get` which is a Numba device function to get the bit based on a given index from a mask bitarray.